core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java

```diff
@@ -1021,7 +1040,14 @@ public <T> CompletionStage<T> invokeCommand(Address target, ReplicableCommand co
       }
       long requestId = requests.newRequestId();
       logRequest(requestId, command, target, "single");
-      SingleTargetRequest<T> request = new SingleTargetRequest<>(collector, requestId, requests, metricsManager.trackRequest(target));
+
+      SingleTargetRequest<T> request;
+      if (tracingEnabled) {
+         request = new SimpleTargetTracedRequest<>(collector, requestId, requests, metricsManager.trackRequest(target),
+               globalTelemetry, () -> createSpan(command));
```

Comment:
don't get why the supplier is required. Also, wouldn't be simple to decorate the Request instance?

Reply:
It was actually the first idea I had to implement the solution.
I changed my mind basically for two reason:
Decorating the Request instance:

1. We have to pollute the Request with fields (tracingEnabled, internalCacheRegistry, cacheManager, globalTelemetry)
2. Basically we have different subtypes of Request (SimpleTargetTracedRequest, MultiTargetTracedRequest, StaggeredTracedRequest)

All in all, I thought that this was a case in which using a lambda would have simplified the impact of the changes on the preexisting code.
I don't know what you think about it.

Reply:
My very concern doing this contribution (that had to deal with a lot of different code areas) is about the fact that I wanted to change as few as possible the preexisting code.

Reply:
I though just a simple decorator, some pseudo-code below.
```java
public class TracingRequest implement Request {
   Request delegate;
   Span span;
   
   void onComplete() {
    span.complete()
    delegate.complete()
   }
}
```
Decorate if tracing is enabled
```java
Request req = new SingleTargetRequest() // or whatever
if (tracing)
  req = new TracingRequest(req, createspan())
```

Reply:
Implements a Request without extends any classes means to implement 42 method I think.